pkg/auth/providers/migration/activedirectory_migration.go

```diff
@@ -96,6 +100,11 @@ func (m *adMigration) migrate() error {
 			if strings.HasPrefix(principal, "activedirectory_user://") && strings.Contains(principal, ",") {
 				err = m.migrateUser(user, principal)
 				if err != nil {
+					var apiError *httperror.APIError
+					if errors.As(err, &apiError) && httperror.IsNotFound(apiError) {
+						logrus.Infof("user %s does not exist, skipping migration", principal)
+						continue
```

Comment:
No, that should not be the case.  The "migration" is only on the Rancher side, no ldap/AD data is changed.  The Refresh code has been adapted to be able to handle either old or new style principals.

Reply:
Are you thinking that this should also be logrus.Trace?  I actually think Info is ok for this one because it is indicative of something actionable on the user's part.  Also, this code should only ever really be ran once for each user and only the first time 2.7.5+ is ran.

> Same as my other comment. But leaving it inline so it can be addressed directly. Can we hit a case where the user is not found because it is in the process of being migrated? Causing the user's data to always be stale.

Are you thinking that this should also be logrus.Trace?  I actually think Info is ok for this one because it is indicative of something actionable on the user's part.  Also, this code should only ever really be ran once for each user and only the first time 2.7.5+ is ran.

Reply:
Oh sorry I was referencing this comment
> My review is more based on the code logic. The overall design appears reasonable though I'd like to know if we can hit a case where the user is migrated during the check, and they get not found, causing us to have stale data until the next refresh occurs.